control/server.py

```diff
@@ -127,10 +133,37 @@ def serve(self):
                 gateway_addr, gateway_port))
 
         # Check for existing NVMeoF target state
-        self._restore_state()
+        callbacks = {
+            self.gateway_state.BDEV_PREFIX: {
+                True: gateway_rpc.create_bdev,
+                False: gateway_rpc.delete_bdev
+            },
+            self.gateway_state.SUBSYSTEM_PREFIX: {
+                True: gateway_rpc.create_subsystem,
+                False: gateway_rpc.delete_subsystem
+            },
+            self.gateway_state.NAMESPACE_PREFIX: {
+                True: gateway_rpc.add_namespace,
+                False: gateway_rpc.remove_namespace
+            },
+            self.gateway_state.HOST_PREFIX: {
+                True: gateway_rpc.add_host,
+                False: gateway_rpc.remove_host
+            },
+            self.gateway_state.LISTENER_PREFIX: {
+                True: gateway_rpc.create_listener,
+                False: gateway_rpc.delete_listener
+            }
+        }
+        self.gateway_state.set_update_callbacks(callbacks)
+        self.gateway_state.update()
         # Start server
         self.logger.info("Starting server...")
         self.server.start()
+        # Start checking for persistent configuration updates
+        self.update_timer = threading.Thread(target=self._update_state)
```

Comment:
Do we want to move the `update_timer` to the consistent gateway state? Since we also update state with notify/update as part of the OMAP gateway state I think this does not belong here.